src/zotero-ocr.js

```diff
@@ -0,0 +1,319 @@
+// zoteroocr.js
+
+// See https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules.
+Components.utils.import("resource://gre/modules/FileUtils.jsm");
+Components.utils.import("resource://gre/modules/osfile.jsm");
+
+
+ZoteroOCR = {
+    id: null,
+    version: null,
+    rootURI: null,
+    initialized: false,
+    addedElementIDs: [],
+    
+    init({ id, version, rootURI }) {
+        if (this.initialized) return;
+        this.id = id;
+        this.version = version;
+        this.rootURI = rootURI;
+        this.initialized = true;
+    },
+    
+    log(msg) {
+        Zotero.debug("ZoteroOCR: " + msg);
+    },
+    
+    addToWindow(window) {
+        let doc = window.document;
+        
+        // Add a stylesheet to the main Zotero pane
+        /*
+        let link1 = doc.createElement('link');
+        link1.id = 'make-it-red-stylesheet';
+        link1.type = 'text/css';
+        link1.rel = 'stylesheet';
+        link1.href = this.rootURI + 'style.css';
+        doc.documentElement.appendChild(link1);
+        this.storeAddedElement(link1);
+        */
+        
+        // Use Fluent for localization
+        window.MozXULElement.insertFTLIfNeeded("zotero-ocr.ftl");
+        
+        // Add menu option
+        let menuitem = doc.createXULElement('menuitem');
+        menuitem.id = 'ocr-selected-pdfs';
+        menuitem.setAttribute('type', 'checkbox');
+        menuitem.setAttribute('data-l10n-id', 'ocr-selected-pdfs');
+        // MozMenuItem#checked is available in Zotero 7
+        menuitem.addEventListener('command', () => {
+            //ZoteroOCR.toggleGreen(window, menuitem.checked);
+            ZoteroOCR.recognize(window);
+        });
+        doc.getElementById('menu_ToolsPopup').appendChild(menuitem);
+        this.storeAddedElement(menuitem);
+    },
+    
+    addToAllWindows() {
+        var windows = Zotero.getMainWindows();
+        for (let win of windows) {
+            if (!win.ZoteroPane) continue;
+            this.addToWindow(win);
+        }
+    },
+    
+    storeAddedElement(elem) {
+        if (!elem.id) {
+            throw new Error("Element must have an id");
+        }
+        this.addedElementIDs.push(elem.id);
+    },
+    
+    removeFromWindow(window) {
+        var doc = window.document;
+        // Remove all elements added to DOM
+        for (let id of this.addedElementIDs) {
+            doc.getElementById(id)?.remove();
+        }
+        doc.querySelector('[href="zotero-ocr.ftl"]').remove();
+    },
+    
+    removeFromAllWindows() {
+        var windows = Zotero.getMainWindows();
+        for (let win of windows) {
+            if (!win.ZoteroPane) continue;
+            this.removeFromWindow(win);
+        }
+    },
+    
+    toggleGreen(window, enabled) {
+        window.document.documentElement
+            .toggleAttribute('data-green-instead', enabled);
+    },
+    
```

Comment:
This code should be removed as it is not used by Zotero OCR.

Reply:
Forgotten code from the make-it-red example plugin, of course it is useless here.